typer/models.py

```diff
@@ -425,7 +425,9 @@ def __init__(
         name: str,
         default: Any = inspect.Parameter.empty,
         annotation: Any = inspect.Parameter.empty,
+        kind: inspect._ParameterKind = inspect.Parameter.POSITIONAL_OR_KEYWORD,
```

Comment:
accessing `inspect._ParameterKind` is ugly a bit, but the type name is `_ParameterKind` 
https://github.com/python/typeshed/blob/a690a14c820238ca7a64cd502343d6b5a649e731/stdlib/inspect.pyi#L125-L134